### ISSUE TITLE
Ability to boot a host with only discovery for agent install

### ISSUE THREAD

**Issue by leelavg** [ozchamo/YAKKO#13]
Hi,

tl:dr; version is
> My dream is to turn the medium box into a CNV node. Now that remote nodes are supported, this is in the short term roadmap, success looks promising but not guaranteed!
- May I know what it takes to have above feature (or what can I learn to have it incorporated)
----
- I came across decade old lab machine(s) (64CPU/64G/1T+3T, pre-configured dhcp) on which I failed to install SNO OCP/OKD (it's been 4 weeks on & off 😅) for dev/testing purposes.
- While I was getting ready to give YAKKO a try, I realised the sol doesn't pick up bridge networks (maybe intentional 🤔).
- I wanted to use a bridge to create SNO on one kvm machine and boot a VM w/ only discovery (~ agent install) on another kvm to work w/ hypershift 
- So is my initial question, if not bridge network how can the sol be used to work w/ hypershift if possible?

Nevertheless, thanks for the all the work, I will definitely try this out.

**Comment by ozchamo** [comment]
Hi ... Lee? :)

Q: Now that remote nodes are supported, this is in the short term roadmap, success looks promising but not guaranteed!
A: YAKKO "for now" has no recourse for remote nodes. You could build that yourself by adding like you say, bridged connectivity to the network, or perhaps a wireguard facility that let's you into the virtual network that YAKKO sets up. Out of the box YAKKO uses HAproxy to "bridge" the virtual network and any hosts connecting to it, BUT HAproxy does not support UDP forwarding, which is an out of box requirement for remote nodes. Nothing should stop you from attempting a cluster build with YAKKO on the node, and then running up a VM as a CNV host. It ain't gonna fly and you don't have a ton of memory either, but... It should work ;)

Q:  I came across decade old lab machine(s) (64CPU/64G/1T+3T, pre-configured dhcp) on which I failed to install SNO OCP/OKD (it's been 4 weeks on & off 😅) for dev/testing purposes.
A: Probably way too old to have RHCOS boot on it... You can always try the Assisted Installer on bare metal, but I doubt that will take off. Kubernetes as we know didn't even quite exist when your box was made, by the looks of it!

Q: While I was getting ready to give YAKKO a try, I realised the sol doesn't pick up bridge networks (maybe intentional 🤔).
A: Correct - YAKKO works solely behind a virtual network, and it users HAproxy as I explained earlier to connect the outside world to the virtual network it builds.

Q: I wanted to use a bridge to create SNO on one kvm machine and boot a VM w/ only discovery (~ agent install) on another kvm to work w/ hypershift
A: That ought to work with the assisted installer and/or the agent based installer. I assume you are running Fedora or RHEL on the box for KVM image creation. Hypershift ... see below.

Q: So is my initial question, if not bridge network how can the sol be used to work w/ hypershift if possible? 
A: I have never attempted to accommodate Hosted Control Planes (AKA Hypershift or HCP). It should not be entirely impossible to add nodes to a HCP cluster within the virtual network, but that's for bigger boys that YAKKO for now.

Now, don't think I don't want to do it, it's been sitting at the top of the code forever, I just need "time" which is scarce these days!
##########################################################################
# DESIRED FEATURES:
# - Add a remote node!!!!
# - (COMPLEX) Work on non-virtual network. That may be a project for the next pandemic
##########################################################################

**Comment by leelavg** [comment]
> Lee? :)
- Leela :)

> BUT HAproxy does not support UDP forwarding, which is an out of box requirement for remote nodes
- ack, good to know this.

> a VM as a CNV host.
- well, I'll mostly end up w/ this.

> too old to have RHCOS boot on it
- yep, learnt the hard way, rhcos didn't boot due to deprecated (~removed) nic drivers
- these machines doesn't allow to attach cdrom but I did progress w/ booting from a drive partition w/ okd4, however hit the dead end when bootkube failed w/ reasons I wasn't able to debug

> Fedora
- I was able to boot these machines w/ fedora 32 but I ran into issues running YAKKO (latest) on this as some steps are restarting libvirtd which is stopping virtnetworkd as well, even after I edited at all places to restart virtnetworkd the sol still fails at defining virtual network
- I upgraded to fedora 39 and now I couldn't ssh into these boxes (kinda expected I guess 😅)

ack for the rest and thanks a lot for the response!
